### PR TITLE
1965 credentials in config

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes part of #1965 submit/reset.

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by exploiting a machine 🥳 
* [x] If applicable, add screenshots or log transcripts of the feature working

image
image
image
image
image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 1]
```diff
@@ -285,21 +323,27 @@ class ConfigurePageComponent extends AuthComponent {
...
     formProperties['liveValidate'] = true;
+    formProperties['formData'] = this.state.currentFormData;
 
     applyUiSchemaManipulators(this.state.selectedSection,
       formProperties['formData'],
       formProperties['uiSchema']);
 
     if (this.state.selectedSection === 'propagation') {
-        return (<PropagationConfig {...formProperties}/>)
+      delete Object.assign(formProperties, {'configuration': formProperties.formData}).formData;
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:339]: Why is this needed?

**Review by mssalvatore — CHANGES_REQUESTED** [review]
I've only gotten half-way through the review, but I don't want to lose my work. More to come.

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx [hunk 2]
```diff
@@ -36,19 +41,22 @@ const ConfigImportModal = (props: Props) => {
...
   useEffect(() => {
-    if (configContents !== null) {
+    if (configContents !== null && configCredentials !== null) {
+      console.log("useEffect was called");
       tryImport();
     }
-  }, [configContents, unsafeOptionsVerified])
+  }, [configContents, configCredentials, unsafeOptionsVerified])
 
   function tryImport() {
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx:50]: I'm doubtful about the flow control of this function. Have you tested the case where config is encrypted and unsafe options are verified? Does it show the unsafe options modal?

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx:50]: Yep, it shows the unsafe options modal.

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 3]
```diff
@@ -144,26 +162,24 @@ class ConfigurePageComponent extends AuthComponent {
...
   }
 
-  onChange = ({formData}) => {
+  onChange = (formData) => {
+    let data = formData;
     let configuration = this.state.configuration;
-    configuration[this.state.selectedSection] = formData;
-    this.setState({currentFormData: formData, configuration: configuration});
+    configuration[this.state.selectedSection] = data;
+    this.setState({currentFormData: data, configuration: configuration});
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:176]: Why are we not just using `formData` here?

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ExportConfigModal.tsx [hunk 4]
```diff
@@ -23,16 +24,17 @@ const ConfigExportModal = (props: Props) => {
 
   function onSubmit() {
     let config = reformatConfig(props.configuration, true);
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ExportConfigModal.tsx:26]: Remove?

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx [hunk 5]
```diff
@@ -61,16 +68,40 @@ const ConfigImportModal = (props: Props) => {
...
       sendConfigToServer();
+      sendConfigCredentialsToServer();
       setUploadStatus(UploadStatuses.success);
     }
   }
 
+  function sendConfigCredentialsToServer() {
+    console.log("sendConfig to server called!!");
+    let credentials = formatCredentialsForIsland(configCredentials);
+    console.log(credentials);
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx:79]: Remove these console log messages

**Review by shreyamalviya — APPROVED** [review]

**Pull request merged**